### DISCUSSION TITLE
How to display a compass needle?

### DISCUSSION THREAD

**Discussion by tve** [Moddable-OpenSource/moddable#1036] [Q&A] [answered]
I'd like to display a compass needle "full screen" on an attached display (well, it's going to be round, so it won't use the full display in the long direction). The compass would be animated to point north as one rotates the unit using a digital compass. I'd like the needle to rotate smoothly, not in something like 10 degree increments. How would I implement this best? I'm not seeing anything that is not a rectangle or a mask/image and it seems to me that I either need arbitrary lines to draw the needle or rotate a mask/image by an arbitrary angle. Maybe I'm missing something or am not thinking about the obvious?

**Comment by mkellner** [comment, marked as answer]
I think you're looking for something like [Outlines](https://github.com/Moddable-OpenSource/moddable/blob/public/documentation/commodetto/outline/Outlines.md).

The [`piu/outline/clock`](https://github.com/Moddable-OpenSource/moddable/tree/public/examples/piu/outline/clock) example has a continuous sweep hand for seconds.

**Reply by phoddie** [reply]
The [blog post](https://blog.moddable.com/blog/scalableoutlines/) has a high level introduction including an example of using Figma to create the outlines.

**Reply by phoddie** [reply]
The Node-RED MCU Edition dashboard implementation includes [gauges](https://github.com/phoddie/node-red-mcu/blob/main/nodes/ui/ui_gauge.js), one of which is a kind of compass. 

https://user-images.githubusercontent.com/6856458/219425782-c5c21acf-d149-4437-8ee9-133dfef0a652.mov

**Reply by tve** [reply]
Thanks for the pointers!
